dask/bag/core.py

```diff
@@ -2498,3 +2479,123 @@ def to_dataframe(seq, columns, dtypes):
         seq = list(seq)
     res = pd.DataFrame(seq, columns=list(columns))
     return res.astype(dtypes, copy=False)
+
+
+def repartition_npartitions(bag, npartitions):
+    """ Changes the number of partitions of the bag.
+
+    This can be used to reduce or increase the number of partitions
+    of the bag.
+    """
+    new_name = "repartition-%d-%s" % (npartitions, tokenize(bag, npartitions))
```

Comment:
Can we move this down a few lines since it might not be needed?

Reply:
also this.

Reply:
whoops i thought I addressed that - i moved it down a couple lines as suggested.